pkg/restore/restore.go

```diff
@@ -2527,3 +2530,41 @@ func (ctx *restoreContext) handleSkippedPVHasRetainPolicy(
 	obj = resetVolumeBindingInfo(obj)
 	return obj, nil
 }
+
+func determineRestoreStatus(obj *unstructured.Unstructured, resourceIncludesExcludes *collections.IncludesExcludes, groupResource string, log logrus.FieldLogger) bool {
+	var shouldRestoreStatus bool
+
+	// Determine restore spec behavior
+	if resourceIncludesExcludes != nil {
+		shouldRestoreStatus = resourceIncludesExcludes.ShouldInclude(groupResource)
+	}
+
+	// Retrieve annotations safely
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		log.Warnf("No restore status specific annotation found for %s, using restore spec setting: %v", kube.NamespaceAndName(obj), shouldRestoreStatus)
```

Comment:
The log is misleading.

This logs when annotation of object is nil, not just because restore status specific annotation is not found.

The same log should be used if annotationExists == false below.

Reply:
updated logging, PTAL, thanks !